src/coreclr/vm/jitinterface.cpp

```diff
@@ -1453,6 +1453,41 @@ static CorInfoHelpFunc getInstanceFieldHelper(FieldDesc * pField, CORINFO_ACCESS
     return (CorInfoHelpFunc)helper;
 }
 
+static OBJECTREF getFrozenBoxedStatic(FieldDesc* field)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    } CONTRACTL_END;
+
+    // These fields never hold frozen boxed statics
+    if (!field->IsStatic() || !field->IsByValue() || field->IsSpecialStatic())
+    {
+        return NULL;
+    }
+
+    TypeHandle typeHandle = field->GetFieldTypeHandleThrowing();
+    MethodTable* pFieldMT = typeHandle.GetMethodTable();
+    if (typeHandle.IsCanonicalSubtype() || pFieldMT->IsSharedByGenericInstantiations() || pFieldMT->ContainsPointers())
```

Comment:
```suggestion
    if (field->GetEnclosingMethodTable()->IsSharedByGenericInstantiations() || pFieldMT->ContainsPointers())
```
I think this should check whether the enclosing type is shared. The actual type of the field should not matter for whether the value is shared or not.